crates/query-engine/translation/src/translation/helpers.rs

```diff
@@ -96,11 +95,11 @@ pub enum CollectionInfo<'env> {
 /// dedicated composite types.
 pub enum CompositeTypeInfo<'env> {
     Table {
-        name: &'env str,
+        name: models::CollectionName,
```

Comment:
Trying to track the references when the types are newtypes becomes difficult, so we don't use references from them. This is probably not a big deal especially because they are SmolStr under the hood.